google-http-client/src/main/java/com/google/api/client/http/javanet/NetHttpTransport.java

```diff
@@ -118,11 +123,21 @@ public NetHttpTransport() {
   NetHttpTransport(ConnectionFactory connectionFactory,
       SSLSocketFactory sslSocketFactory, HostnameVerifier hostnameVerifier) {
     this.connectionFactory =
-        connectionFactory == null ? new DefaultConnectionFactory() : connectionFactory;
+        getConnectionFactory(connectionFactory);
     this.sslSocketFactory = sslSocketFactory;
     this.hostnameVerifier = hostnameVerifier;
   }
 
+  private ConnectionFactory getConnectionFactory(ConnectionFactory connectionFactory) {
+    if (connectionFactory == null) {
+      if (System.getProperty("com.api.client.should_use_proxy") != null) {
```

Comment:
this should be com.google.api.client.should_use_proxy

Reply:
Thanks @coekie . Quick followup PR: https://github.com/google/google-http-java-client/pull/383/files